flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/factories/datagen/DataGenVisitorBase.java

```diff
@@ -0,0 +1,110 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.factories.datagen;
+
+import org.apache.flink.api.common.functions.RuntimeContext;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.runtime.state.FunctionInitializationContext;
+import org.apache.flink.streaming.api.functions.source.datagen.DataGenerator;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.types.logical.DateType;
+import org.apache.flink.table.types.logical.LocalZonedTimestampType;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.TimeType;
+import org.apache.flink.table.types.logical.TimestampType;
+import org.apache.flink.table.types.logical.ZonedTimestampType;
+import org.apache.flink.table.types.logical.utils.LogicalTypeDefaultVisitor;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.util.function.Supplier;
+
+/**
+ * Base class for translating {@link LogicalType LogicalTypes} to {@link DataGeneratorContainer}'s.
+ */
+public abstract class DataGenVisitorBase extends LogicalTypeDefaultVisitor<DataGeneratorContainer> {
+
+	protected final String name;
+
+	protected final ReadableConfig config;
+
+	protected DataGenVisitorBase(String name, ReadableConfig config) {
+		this.name = name;
+		this.config = config;
+	}
+
+	@Override
+	public DataGeneratorContainer visit(DateType dateType) {
+		return DataGeneratorContainer.of(TimeGenerator.of(LocalDate::now));
+	}
+
+	@Override
+	public DataGeneratorContainer visit(TimeType timeType) {
+		return DataGeneratorContainer.of(TimeGenerator.of(LocalTime::now));
+	}
+
+	@Override
+	public DataGeneratorContainer visit(TimestampType timestampType) {
+		return DataGeneratorContainer.of(TimeGenerator.of(LocalDateTime::now));
+	}
+
+	@Override
+	public DataGeneratorContainer visit(ZonedTimestampType zonedTimestampType) {
+		return DataGeneratorContainer.of(TimeGenerator.of(OffsetDateTime::now));
+	}
+
+	@Override
+	public DataGeneratorContainer visit(LocalZonedTimestampType localZonedTimestampType) {
+		return DataGeneratorContainer.of(TimeGenerator.of(Instant::now));
+	}
+
+	@Override
+	protected DataGeneratorContainer defaultMethod(LogicalType logicalType) {
+		throw new ValidationException("Unsupported type: " + logicalType);
+	}
+
+	private interface SerializableSupplier<T> extends Supplier<T>, Serializable { }
```

Comment:
I think lambda is already `Serializable`.

Reply:
Its not, the source gets caught in the closure cleaner without this.

Reply:
I meant we could do something like `return (Supplier<FileSystem> & Serializable) () -> {...}`, but it is OK for now.